File: app/src/main/java/com/task/ui/component/news/HomeContract.java
package com.task.ui.component.news;

import com.task.data.remote.dto.NewsItem;
import com.task.ui.base.listeners.BaseView;
import com.task.ui.base.listeners.RecyclerItemListener;

import java.util.List;

/**
 * Created by AhmedEltaher on 5/12/2016
 */

public interface HomeContract {

    interface View extends BaseView {
        void initializeNewsList(List<NewsItem> news);

        void setLoaderVisibility(boolean isVisible);

        void navigateToDetailsScreen(NewsItem news);

        void setNoDataVisibility(boolean isVisible);

        void setListVisibility(boolean isVisible);

        void showSearchError();

        void showNoNewsError();

        void incrementCountingIdlingResource();

        void decrementCountingIdlingResource();
    }


    interface Presenter {
        void getNews();

        void onSearchClick(String newsTitle);

        void unSubscribe();

        RecyclerItemListener getRecyclerItemListener();
    }
}

File: app/src/main/java/com/task/ui/component/news/HomeContract.kt
package com.task.ui.component.news

import com.task.data.remote.dto.NewsItem
import com.task.ui.base.listeners.BaseView
import com.task.ui.base.listeners.RecyclerItemListener

/**
 * Created by AhmedEltaher on 5/12/2016
 */

interface HomeContract {

    interface View : BaseView {
        fun initializeNewsList(news: List<NewsItem>)

        fun setLoaderVisibility(isVisible: Boolean)

        fun navigateToDetailsScreen(news: NewsItem)

        fun setNoDataVisibility(isVisible: Boolean)

        fun setListVisibility(isVisible: Boolean)

        fun showSearchError()

        fun showNoNewsError()

        fun incrementCountingIdlingResource()

        fun decrementCountingIdlingResource()
    }


    interface Presenter {

        fun getRecyclerItemListener():RecyclerItemListener

        fun getNews()

        fun onSearchClick(newsTitle: String)

        fun unSubscribe()
    }
}
